junit/src/test/java/cucumber/runtime/junit/FeatureRunnerTest.java

```diff
@@ -89,17 +93,67 @@ public void should_call_formatter_for_scenario_outline_with_two_examples_table_a
     }
 
     private RunNotifier runFeatureWithNotifier(CucumberFeature cucumberFeature) throws InitializationError {
+        FeatureRunner runner = createFeatureRunner(cucumberFeature);
+        RunNotifier notifier = mock(RunNotifier.class);
+        runner.run(notifier);
+        return notifier;
+    }
+
+    private FeatureRunner createFeatureRunner(CucumberFeature cucumberFeature) throws InitializationError {
         final RuntimeOptions runtimeOptions = new RuntimeOptions("-p null");
         final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
         final ClasspathResourceLoader resourceLoader = new ClasspathResourceLoader(classLoader);
         final RuntimeGlue glue = mock(RuntimeGlue.class);
         final Runtime runtime = new Runtime(resourceLoader, classLoader, asList(mock(Backend.class)), runtimeOptions, new TimeService.Stub(0l), glue);
-        FeatureRunner runner = new FeatureRunner(cucumberFeature, runtime, new JUnitReporter(runtime.getEventBus(), false, new JUnitOptions(Collections.<String>emptyList())));
-        RunNotifier notifier = mock(RunNotifier.class);
-        runner.run(notifier);
-        return notifier;
+        return new FeatureRunner(cucumberFeature, runtime, new JUnitReporter(runtime.getEventBus(), false, new JUnitOptions(Collections.<String>emptyList())));
+    }
+
+
+    @Test
+    public void shouldPopulateDescriptionsWithStableUniqueIds() throws Exception {
+        CucumberFeature cucumberFeature = TestPickleBuilder.parseFeature("path/test.feature", "" +
+            "Feature: feature name\n" +
+            "  Background:\n" +
+            "    Given background step\n" +
+            "  Scenario: A\n" +
+            "    Then scenario name\n" +
+            "  Scenario: B\n" +
+            "    Then scenario name\n" +
+            "  Scenario Outline: C\n" +
+            "    Then scenario <name>\n" +
+            "  Examples:\n" +
+            "    | name |\n" +
+            "    | C    |\n" +
+            "    | D    |\n" +
+            "    | E    |\n"
+
+        );
+
+        FeatureRunner runner = createFeatureRunner(cucumberFeature);
+        FeatureRunner rerunner = createFeatureRunner(cucumberFeature);
+
+        Set<Description> descriptions = new HashSet<Description>();
+        assertDescriptionIsUnique(runner.getDescription(), descriptions);
```

Comment:
It is a bit unusual for an assert-method to modify its arguments, but checking the uniqueness with a set makes sence - and then the set has been create and can be used subsequently.